### PR TITLE
Python setup cleanup

### PR DESCRIPTION
**Pull request by flit**
- Removed requirements files; all requirements were moved into `setup()`.
- Added missing things to setup.
- Removed (incorrect) version and author from the Rust CLI.
- Added `*.tags` to `.gitignore`.

### CODE REVIEW

**Comment by mathias-arm** [comment]
The `build-wheels.sh` script relies on the requirements `.txt` file for CI.

**Comment by mathias-arm** [comment]
Can you also include an update to `.github/workflows/ci.yml`? `build-wheels.sh` either needs to be updated as well or the whole Travis CI setup needs to be removed.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -38,15 +38,12 @@ jobs:
           find ./ -name "__pycache__" -type d -print0 |xargs -0 rm -rf
           find ./ -name "*.pyc" -type f -print0 |xargs -0 rm -rf
           find ./ -name "*.so" -type f -print0 |xargs -0 rm -rf
-          pip install -U setuptools pip
-          pip install -U --upgrade-strategy=eager -r requirements.txt
-          pip install -U --upgrade-strategy=eager -r setup_requirements.txt
+          pip install -U setuptools pip wheel
           pip install -U --upgrade-strategy=eager -r test_requirements.txt
```
Comment (mathias-arm) [inline .github/workflows/ci.yml:42]: Can we get rid of `test_requirements.txt `?

There is a solution by adding `extras_require` with a `test` entry (with same value as `tests_require`).  The `pip install -e '.[test]'` command can be used.

Comment (flit) [reply .github/workflows/ci.yml:42]: Sure.

Fyi, I'm not planning to merge this until I move to PEP517. I'll change it to a draft.

**Pull request merged**